src/pool/srv_util.c

```diff
@@ -452,8 +438,24 @@ ds_pool_map_tgts_update(struct pool_map *map, struct pool_target_id_list *tgts,
 				    &version);
 		if (rc != 0)
 			return rc;
-	}
 
+		if (evict_rank &&
+		    !(dom->do_comp.co_status & (PO_COMP_ST_DOWN |
+						PO_COMP_ST_DOWNOUT)) &&
+		    pool_map_node_status_match(dom, PO_COMP_ST_DOWN |
+					       PO_COMP_ST_DOWNOUT)) {
+			if (opc == POOL_EXCLUDE)
+				dom->do_comp.co_status = PO_COMP_ST_DOWN;
+			else if (opc == POOL_EXCLUDE_OUT)
+				dom->do_comp.co_status = PO_COMP_ST_DOWNOUT;
+			else
+				D_ASSERTF(false, "evict rank by %d\n", opc);
+			dom->do_comp.co_fseq = target->ta_comp.co_fseq;
+			D_DEBUG(DF_DSMS, "change rank %u to DOWN\n",
+				dom->do_comp.co_rank);
+			version++;
```

Comment:
Is version bumping intentional? I suppose an additional bumping (one for target state change, one for domain state change) won't cause trouble, but I'm not sure if bumping version for domain state change will cause trouble? Imagine all targets are already in DOWN or DOWNOUT before, then this operation will change the domain state only.

Reply:
I can not think of any throuble of bumping version, but it will help us to update the pool map, so to broadcast the domain status to all servers by IV.  also make rebuild tracking easier This patch is actually to fix one of rebuild issue we saw.